stream_alert/rule_processor/threat_intel.py

```diff
@@ -164,14 +164,20 @@ def _extract_ioc_from_record(self, record):
                         for original_key in original_keys:
                             value = value[original_key]
                     if value:
-                        # Threat Intel will only check against public IP addresses
-                        # while processing IP IOCs.
-                        if ioc_type == 'ip' and not self.is_public_ip(value):
+                        # Threat Intel will skip searching for known excluded IOCs
+                        if self.is_excluded_ioc(ioc_type, value):
                             continue
                         ioc_value_type_tuples.add((value, ioc_type))
         return [StreamIoc(value=str(value).lower(), ioc_type=ioc_type, associated_record=record)
                 for value, ioc_type in ioc_value_type_tuples]
 
+    def _setup_excluded_iocs(self, config_excluded_iocs=None):
+        if config_excluded_iocs:
+            self.excluded_iocs = config_excluded_iocs
```

Comment:
This function sets the `excluded_iocs` instance property from a call to `_setup_excluded_iocs` within the `__init__`. This can lead to confusion, since the property is not defined within the `__init__` itself (and could result in users/maintainers not realizing the property exists.

What I would suggest is have this function _return_ the `excluded_iocs` that would then get set on `self` within the init.. for example:


```
def __init__(self, excluded_iocs, table, region='us-east-1'):
    self.excluded_iocs = self._setup_excluded_iocs(excluded_iocs)

...

def _setup_excluded_iocs(self, config_excluded_iocs=None):
    if not config_excluded_iocs:
        return {'ip': set(), 'domain': set(), 'md5': set()}

    # Transform the IPs into IPNetwork objects
    config_excluded_iocs['ip'] = {IPNetwork(ip) for ip in config_excluded_iocs.get('ip', [])}
    return config_excluded_iocs
```

This way all instance properties are defined within `__init__` as opposed to within the _scope_ of `__init__`